core/src/main/java/org/apache/iceberg/rest/HTTPClient.java

```diff
@@ -319,11 +322,9 @@ protected <T extends RESTResponse> T execute(
 
   @Override
   public void close() throws IOException {
-    try {
-      if (authSession != null) {
```

Comment:
when is the auth session being closed then? I currently don't see any code that would close the auth session anywhere or am I missing something?

Reply:
The auth session being provided externally, it's the caller's responsibility to close it now.